tests/test_samples.py

```diff
@@ -86,28 +89,45 @@ def test_read_multinest():
     ns = NestedSamples(root='./tests/example_data/mn_old')
     ns.plot_2d(['x0', 'x1', 'x2', 'x3'])
     ns.plot_1d(['x0', 'x1', 'x2', 'x3'])
+    plt.close("all")
 
 
 def test_read_polychord():
     ns = NestedSamples(root='./tests/example_data/pc')
     ns.plot_2d(['x0', 'x1', 'x2', 'x3'])
     ns.plot_1d(['x0', 'x1', 'x2', 'x3'])
+    plt.close("all")
 
 
 def test_plot_2d_types():
     ns = NestedSamples(root='./tests/example_data/pc')
     params_x = ['x0', 'x1', 'x2', 'x3']
     params_y = ['x0', 'x1', 'x2']
-    fig, axes = ns.plot_2d([params_x, params_y], types=['kde', 'scatter'])
-    assert(axes.isnull().sum().sum() == 0)
+    params = [params_x, params_y]
+    # Test old interface
+    fig, axes = ns.plot_2d(params, types=['kde', 'scatter'])
+    assert((~axes.isnull()).sum().sum() == 12)
```

Comment:
In `test_make_2d_axes` you define `nx = len(paramnames_x)` and use those to determine the numeric value to test for in the assertions. Would probably be more appropriate here, too, right?

Reply:
This turned out to be pretty challenging in practice to do, so I've re-written those sections of the tests to be more systematic

Reply:
Looks great, now, and seems to cover many of cases.